fastapi_async_langchain/callback.py

```diff
@@ -18,3 +18,9 @@ def always_verbose(self) -> bool:
     async def on_llm_new_token(self, token: str, **kwargs: Any) -> None:
         """Run on new LLM token. Only available when streaming is enabled."""
         await self.send(token)
+
+    async def on_chain_end(self, outputs: Dict[str, Any], **kwargs: Any) -> None:
+        """Run when chain ends running."""
+        if not outputs['source_documents'] is None:
+            source_documents = '\n'.join(f"{doc.page_content}: {doc.metadata['source']}" for doc in outputs['source_documents'])
+            await self.send("\n\nSOURCE DOCUMENTS: \n" + source_documents)
```

Comment:
lgtm! i am wondering if we should create callback classes per chain type, to avoid bloating a single class. what do you think?

Reply:
It seems good to me.  

I tried to work as you propose I hope it is as you had thought.